crates/meilisearch/tests/search/facet_search.rs

```diff
@@ -221,8 +221,15 @@ async fn add_documents_and_deactivate_facet_search() {
     let (response, code) =
         index.facet_search(json!({"facetName": "genres", "facetQuery": "a"})).await;
 
-    assert_eq!(code, 200, "{}", response);
-    assert_eq!(dbg!(response)["facetHits"].as_array().unwrap().len(), 0);
+    assert_eq!(code, 400, "{}", response);
+    snapshot!(response, @r###"
+    {
+      "message": "The facet search is disabled for this index",
+      "code": "invalid_facet_search_disabled",
```

Comment:
Minor suggestion that might not fit the actual naming strategy in other parts of the program, but I find `invalid` a bit out of place here.

As in, `invalid_search_vector` makes sense because it indicates the search vector itself is not valid. `invalid_facet_search_disabled` is less clear—what exactly is invalid here?

That said, it could also be read as "Invalid: facet search disabled", I guess, but I'm not sure there's precedent in the existing code.

```suggestion
      "code": "facet_search_disabled",
```

Reply:
If we are seeking consistency, note that we already have the `feature_not_enabled` error code when trying to use an experimental feature without enabling it first.

Here it might make sense to have `facet_search_disabled` though, because the default state of facet search is enabled (so the issue is that it was disabled), while the default state of exp features is disabled (so the issue is that it was not enabled).

In summary, I support `facet_search_disabled` here.